pants

```diff
@@ -228,8 +228,8 @@ function bootstrap_virtualenv {
         die "SHA256 of ${VIRTUALENV_ZIPAPP_URL} is not as expected. Aborting."
       fi
       green "SHA256 fingerprint of ${VIRTUALENV_ZIPAPP_URL} verified."
-      ln -s "${staging_dir}/${VIRTUALENV_ZIPAPP}" "${staging_dir}/latest"
-      mv "${staging_dir}/latest" "${bootstrapped}"
+      mv -f "${staging_dir}/${VIRTUALENV_ZIPAPP}" "${bootstrapped}"
```

Comment:
I just noticed this but `VIRTUALENV_ZIPAPP` is just the unqualified `virtualenv.pyz`; so bumping the version we download / changing the expected sha256 going forward will no-op above on line 218. So that needs a fix. At that point it may make sense to house the downloaded `virtualenv.pyz` in a directory with its version or hash or both as part of the naming scheme. Then here make a hard link instead of doing a mv.
This way multiple clones of different pants-using repos won't thrash the cache.

Reply:
Oh, good catch. Not sure why we need the hard link, if the final resting place of the file, as returned from this function, includes the version? This mv and rmdir is just to clean up the tmpdir. See most recent commit for a fix for this issue, does that seem reasonable?